### PR TITLE
Increase branch Coverage

### LINKED ISSUE #87: [4337 Module] Increase test coverage to 99+%
- Update hardhat version
- All branches need to be covered
- (optional) Add separate tests using latest bytecode for entrypoint on official address

### PR DESCRIPTION
**Pull request by nlordell**
Partially fixes #87

This PR adds unit tests to cover the remaining branches from the `SimpleEIP4337Module` contract. With this, we should have 100% coverage for the core contracts in the repo (although, not all paths in the test contracts are covered, which should be fine).

In terms of test organisation, I added a `EIP4337Module.spec.ts` for unit tests directly targeting the module functions (`validateUserOp` and `execUserOp*`).

### CODE REVIEW

File: 4337/test/eip4337/EIP4337Module.spec.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import {
+  buildSafeUserOp,
+  calculateSafeOperationHash,
+  buildUserOperationFromSafeUserOperation,
+  buildSafeUserOpTransaction,
+  signSafeOp,
+} from '../../src/utils/userOp'
+import { chainId } from '../utils/encoding'
+
+describe('Simple4337Module', async () => {
```
Comment (mmv08) [inline 4337/test/eip4337/EIP4337Module.spec.ts:15]:
```suggestion
describe('Simple4337Module', () => {
```
This doesn't need to be async anymore

File: 4337/test/eip4337/EIP4337ModuleExisting.spec.ts [hunk 2]
```diff
@@ -128,6 +127,22 @@ describe('EIP4337Module - Existing Safe', async () => {
       expect(emittedRevert).to.be.true
     })
 
+    it('executeUserOpWithErrorString should execute contract calls', async () => {
```
Comment (mmv08) [inline 4337/test/eip4337/EIP4337ModuleExisting.spec.ts:130]: One thing I actually forgot to do in the `executeUserOpWithErrorString` PR is to add tests for it to the EIP4337ModuleNew.spec.ts 🙈 would be amazing if you add them

**Review by mmv08 — APPROVED** [review]
Looks good, just two minor suggestions

File: 4337/test/eip4337/EIP4337Module.spec.ts (hunk 1 above)
Comment (nlordell) [reply 4337/test/eip4337/EIP4337Module.spec.ts:15]: Nice catch - can probably be removed everywhere actually, now that the `let [user] = await ethers.getSigners()` moved to the async `setup` function.

File: 4337/test/eip4337/EIP4337ModuleExisting.spec.ts (hunk 2 above)
Comment (nlordell) [reply 4337/test/eip4337/EIP4337ModuleExisting.spec.ts:130]: Added - the revert propagation test was also missing which I added (asserting that the Safe gets deployed even if the user op reverts).

File: 4337/test/eip4337/EIP4337Module.spec.ts (hunk 1 above)
Comment (nlordell) [reply 4337/test/eip4337/EIP4337Module.spec.ts]: 300395be039b3963834216463be2cab93705ef1d

File: 4337/test/eip4337/EIP4337ModuleExisting.spec.ts (hunk 2 above)
Comment (nlordell) [reply 4337/test/eip4337/EIP4337ModuleExisting.spec.ts:130]: 4f8cd914bbd7a3fd86c4a4506f1d1b2f6b052e5c

**Pull request merged**